applovin_max/android/src/main/java/com/applovin/applovin_max/AppLovinMAXAdView.java

```diff
@@ -17,24 +13,82 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import io.flutter.plugin.common.BinaryMessenger;
-import io.flutter.plugin.common.MethodCall;
 import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.Result;
 import io.flutter.plugin.platform.PlatformView;
 
 /**
  * Created by Thomas So on July 17 2022
  */
 public class AppLovinMAXAdView
-        implements PlatformView, MaxAdViewAdListener, MaxAdRevenueListener
+        implements PlatformView
 {
-    private static final Map<String, MaxAdView> adViewInstances = new HashMap<>( 2 );
+    private static final Map<String, AppLovinMAXAdViewPlatformWidget> platformWidgetInstances          = new HashMap<>( 2 );
+    private static final Map<String, AppLovinMAXAdViewPlatformWidget> preloadedPlatformWidgetInstances = new HashMap<>( 2 );
+
+    @Nullable
+    private AppLovinMAXAdViewPlatformWidget platformWidget;
 
     private final MethodChannel channel;
-    private final MaxAdView     adView;
 
     public static MaxAdView getInstance(final String adUnitId)
     {
-        return adViewInstances.get( adUnitId );
+        AppLovinMAXAdViewPlatformWidget platformWidget = preloadedPlatformWidgetInstances.get( adUnitId );
+        if ( platformWidget == null ) platformWidget = platformWidgetInstances.get( adUnitId );
+        return ( platformWidget != null ) ? platformWidget.getAdView() : null;
```

Comment:
This is kind of hard to read, can we do:

```
    return Optional.ofNullable( preloadedPlatformWidgetInstances.get( adUnitId ) )
                   .orElse( platformWidgetInstances.get( adUnitId ) )
                   .getAdView();
```

Reply:
This requires API 24, but our min is 16.  Can we just leave it as is?

Reply:
Thanks for checking, I had a hunch :). Yeah let's leave as is.